File: src/Benchmarking.java
public class Benchmarking {
    MetodosOrdenamiento metodosOrdenamiento = new MetodosOrdenamiento();

    public Benchmarking() {
        MetodosOrdenamiento metodosOrdenamiento = new MetodosOrdenamiento();
        int[] arreglo = generarArregloAleatorio(10000);
        Runnable tarea = () -> metodosOrdenamiento.burbujaTradicional(arreglo.clone());

        double tiempoNano = medirConNanoTime(tarea);
        double tiempoMillis = medirConCurrentTime(tarea);

        System.out.println("Tiempo medido con nanoTime: " + tiempoNano + " segundos");
        System.out.println("Tiempo medido con currentTimeMillis: " + tiempoMillis + " segundos");
    }


    // Método para generar arreglo aleatorio
    public int[] generarArregloAleatorio(int tam) {
        int[] arreglo = new int[tam];
        for (int i = 0; i < tam; i++) {
            arreglo[i] = (int)(Math.random() * 100000);
        }
        return arreglo;
    }

    // Tiempo usando nanoTime (en segundos)
    public double medirConNanoTime(Runnable tarea) {
        long inicio = System.nanoTime();
        tarea.run();
        long fin = System.nanoTime();
        return (fin - inicio) / 1_000_000_000.0;
    }

    // Tiempo usando currentTimeMillis (en segundos)
    public double medirConCurrentTime(Runnable tarea) {
        long inicio = System.currentTimeMillis();
        tarea.run();
        long fin = System.currentTimeMillis();
        return (fin - inicio) / 1000.0;
    }

    // Método main
    public static void main(String[] args) {
        new Benchmarking();
    }
}

File: src/src_phyton/benchmarking.py
import random
import time
from metodos_ordenamiento import MetodosOrdenamiento

class Benchmarking:
    def __init__(self):
        print('Bench inicializado')
        self.mOrdenamiento = MetodosOrdenamiento()
        arreglo = self.build_arreglo(1000)

        tarea = lambda: self.mOrdenamiento.sortByBubble(arreglo)  # Crear función anónima
        tiempoMillis = self.contar_con_current_time_millis(tarea)
        tiempoNano = self.contar_con_nano_time(tarea)

        print("Tiempo en milisegundos:", tiempoMillis, "ms")
        print("Tiempo en nanosegundos:", tiempoNano, "s")

    def build_arreglo(self, size):  # Crear otro método para construir arreglo
        array = []
        for i in range(size):
            num = random.randint(0, 99999)
            array.append(num)
        return array

    def contar_con_current_time_millis(self, tarea):
        inicio = time.time()
        tarea()
        fin = time.time()
        return (fin - inicio) * 1000

    def contar_con_nano_time(self, tarea):
        inicio = time.time_ns()
        tarea()
        fin = time.time_ns()
        return (fin - inicio) / 1_000_000_000.0

